### PR TITLE
[mysql_enterprise] - update package-spec to 2.9.0

### PR DESCRIPTION
**Pull request by taylor-swanson**
## What does this PR do?

- Update package spec to 2.9.0

```
[git-generate]
go run github.com/andrewkroh/go-examples/ecs-update@latest -ecs-version=8.8.0 -ecs-git-ref=v8.8.0 -format-version=2.9.0 packages/mysql_enterprise
```

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Relates elastic/security-team#5870

### CODE REVIEW

File: packages/mysql_enterprise/data_stream/audit/sample_event.json [hunk 1]
```diff
@@ -61,7 +62,7 @@
         "file": {
             "path": "/tmp/service_logs/mysql_audit.log"
         },
-        "offset": 462
+        "offset": 0
```
Comment (efd6) [inline packages/mysql_enterprise/data_stream/audit/sample_event.json:65]: Why have these zeroed?

Comment (taylor-swanson) [reply packages/mysql_enterprise/data_stream/audit/sample_event.json:65]: I think it's a different event being used for the sample event. This was produced from the system tests via `elastic-package test -g`

Comment (taylor-swanson) [reply packages/mysql_enterprise/data_stream/audit/sample_event.json:65]: Should correspond to the first event here: https://github.com/elastic/integrations/blob/main/packages/mysql_enterprise/_dev/deploy/docker/sample_logs/mysql_audit.log#L1

Comment (efd6) [reply packages/mysql_enterprise/data_stream/audit/sample_event.json:65]: Yep. Confirmed locally that events are emitted with non-zero offsets.

**Review by efd6 — APPROVED** [review]

**Comment by efd6** [comment]
I stopped watching the tests too early, running this on v8.8.2 I see this:
```
FAILURE DETAILS:
mysql_enterprise/audit logfile:
[0] parsing field value failed: field "event.type" value "connection" is not one of the expected values (access, change, info, error) for any of the values of "event.category" (database)
[1] parsing field value failed: field "event.type" value "connection" is not one of the expected values (access, change, info, error, admin, creation, deletion, group, user) for any of the values of "event.category" (database, iam)
```

**Comment by taylor-swanson** [comment]
> I stopped watching the tests too early, running this on v8.8.2 I see this:
> 
> ```
> FAILURE DETAILS:
> mysql_enterprise/audit logfile:
> [0] parsing field value failed: field "event.type" value "connection" is not one of the expected values (access, change, info, error) for any of the values of "event.category" (database)
> [1] parsing field value failed: field "event.type" value "connection" is not one of the expected values (access, change, info, error, admin, creation, deletion, group, user) for any of the values of "event.category" (database, iam)
> ```

That is... bad. Our tests aren't catching this. I'll take a look at it.

**Comment by efd6** [comment]
The `kibana.version` is "^7.17.0 || ^8.0.0" so the tests are being run on 7.17.0, which doesn't have any concerns about these constraints.

**Comment by taylor-swanson** [comment]
> The `kibana.version` is "^7.17.0 || ^8.0.0" so the tests are being run on 7.17.0, which doesn't have any concerns about these field value constraints.

That makes sense for CI, but it's also passing for me locally on 8.9.0 and I'm not sure why. Looking at the expected files, it should be failing.

**Review by efd6 — APPROVED** [review]

**Pull request merged**